x-pack/plugins/apm/public/components/app/transaction_details/waterfall_with_summary/waterfall_container/waterfall/span_flyout/index.tsx

```diff
@@ -85,39 +87,39 @@ const ContainerWithMarginRight = euiStyled.div`
 `;
 
 interface Props {
-  span?: Span;
-  parentTransaction?: Transaction;
+  spanId: string;
+  parentTransactionId?: string;
+  traceId: string;
   totalDuration?: number;
-  onClose: () => void;
   spanLinksCount: SpanLinksCount;
+  onClose: () => void;
 }
 
+const INITIAL_DATA = {
+  span: undefined,
+  parentTransaction: undefined,
+};
+
 export function SpanFlyout({
-  span,
-  parentTransaction,
+  spanId,
+  parentTransactionId,
+  traceId,
   totalDuration,
   onClose,
   spanLinksCount,
 }: Props) {
-  if (!span) {
-    return null;
-  }
+  const { data = INITIAL_DATA, status } = useFetcher(
+    (callApmApi) => {
+      return callApmApi('GET /internal/apm/traces/{traceId}/spans/{spanId}', {
+        params: { path: { traceId, spanId }, query: { parentTransactionId } },
+      });
+    },
+    [traceId, spanId, parentTransactionId]
+  );
 
-  const stackframes = span.span.stacktrace;
-  const codeLanguage = parentTransaction?.service.language?.name;
-  const spanDb = span.span.db;
-  const spanTypes = getSpanTypes(span);
-  const spanHttpStatusCode =
-    span.http?.response?.status_code || span.span?.http?.response?.status_code;
-  const spanHttpUrl = span.url?.original || span.span?.http?.url?.original;
-  const spanHttpMethod = span.http?.request?.method || span.span?.http?.method;
+  const { span, parentTransaction } = data;
 
-  const spanLinksTabContent = getSpanLinksTabContent({
-    spanLinksCount,
-    traceId: span.trace.id,
-    spanId: span.span.id,
-    processorEvent: ProcessorEvent.span,
-  });
+  const isLoading = isPending(status);
```

Comment:
Yay, I didn't see that you added `isPending`. Thanks!

Reply:
Yeah, I'm rendering the loading component down there because I want it to be shown inside the flyout, so I first need to open it.